### PR TITLE
some code cleanups and python 3 adjustments

### PR DESCRIPTION
**Pull request by marksmayo**
cleaned up imports
replaced format strings with f-strings
removed trailing newlines
replaced exit() with sys.exit()
py3 doesn't need inheritance from object, fixed a few of those
replaced a deprecated call to warn() with warning()
fixed a bunch of flake8 warnings (indicated in that commit)

pytest run, I'm actually getting failures, but the same ones that failed before I started these changes, so considering it ok for now.

### CODE REVIEW

**Comment by gaogaotiantian** [comment]
Thanks for taking the effort to do this!

I have a couple of comments, the things that I did not mention, I liked them and they'll be accepted.
1. Are you using `flake8` directly or did you add some extensions? `flake8` is what the github action lint uses so there should not be many warnings. If there are some configurations that you recommend, you can update it in `.github/workflows/lint.yaml`.
2. For trailing commas, I'm okay for those where there's a possibility to extend and there are multiple elements. However, I found it really weird for adding a trailing comma for a single item(or single line). Sometimes I used multiline just to avoid having a really long line. I hope that you can revert changes for the single item ones.
3. Please do not change anything in `src/viztracer/attach_process` and `src/viztracer/web_dist`, that's 3rd party code. The code in `tests/archived` will be deprecated soon so do not change anything in it either. Also, leave `docs/conf.py` alone.
4. You removed `ModuleNotFoundError`, then that except does not need the parenthesis anymore.

Thank you agian for working on this!

**Comment by marksmayo** [comment]
1) Flake8, I have Installed plugins: flake8-2020: 1.7.0, flake8-bandit: 4.1.1, flake8-bugbear: 23.3.23, flake8-builtins: 2.1.0, flake8-commas: 2.1.0, flake8-comprehensions: 3.12.0, flake8-datetimez: 20.10.0, flake8-docstrings:
1.7.0, flake8-mutable: 1.2.0, flake8-noqa: 1.3.1, flake8-pie: 0.16.0, flake8-pytest-style: 1.7.2, flake8-return: 1.2.0, flake8_simplify: 0.20.0, mccabe: 0.7.0, pep8-naming: 0.13.3, pycodestyle: 2.10.0,
pyflakes: 3.0.1 installed.  Thoughts?
2) this is based on flake8-commas, and recommendations from more recent versions of python standards...up to you?
3. fair, I'll undo
4. good point!

**Comment by gaogaotiantian** [comment]
The plugins are too much. I think the official flake8 works for me at this point.

I'll take a look at all the trailing comma fixes and see if there's anything that I hate, maybe it can be rewritten in another way.

File: src/viztracer/__main__.py [hunk 1]
```diff
@@ -3,6 +3,5 @@
 
 from .main import main
 
-
```
Comment (gaogaotiantian) [inline src/viztracer/__main__.py:6]: The style of blank lines between import and the code is inconsistent in CPython repo itself. However, I'd like to keep 2 lines, so let's revert all the line reductions(except for the line between `__all__` and import, I can consider `__all__` as part of import)

File: src/viztracer/viewer.py [hunk 2]
```diff
@@ -17,12 +16,12 @@
...
 import threading
 import time
-from typing import Any, Callable, Dict, List, Optional
 import urllib.parse
+from http import HTTPStatus
+from typing import Any, Callable, Dict, List, Optional
 
 from .flamegraph import FlameGraph
 
-
```
Comment (gaogaotiantian) [inline src/viztracer/viewer.py:25]: 2 lines here.

File: src/viztracer/vizevent.py [hunk 3]
```diff
@@ -4,7 +4,6 @@
 
 from typing import TYPE_CHECKING
 
-
```
Comment (gaogaotiantian) [inline src/viztracer/vizevent.py:7]: This is fine, part of import.

File: src/viztracer/vizplugin.py [hunk 4]
```diff
@@ -3,11 +3,10 @@
 
 
 import sys
-from typing import Dict, Optional, Sequence, Union, TYPE_CHECKING
+from typing import TYPE_CHECKING, Dict, Optional, Sequence, Union
 
 from . import __version__
-from .util import compare_version, color_print
-
```
Comment (gaogaotiantian) [inline src/viztracer/vizplugin.py:10]: This is fine too

File: tests/archived/_test_simulator.py [hunk 5]
```diff
@@ -2,13 +2,12 @@
...
 
 import os
-import sys
 import subprocess
+import sys
 
-from .package_env import package_matrix
-from .util import get_json_file_path, adapt_json_file, generate_json
 from .base_tmpl import BaseTmpl
-
```
Comment (gaogaotiantian) [inline tests/archived/_test_simulator.py:11]: two lines here

File: tests/base_tmpl.py [hunk 6]
```diff
@@ -9,10 +9,9 @@
 import time
 from unittest import TestCase
 
-
```
Comment (gaogaotiantian) [inline tests/base_tmpl.py:12]: two here

File: tests/cmdline_tmpl.py [hunk 7]
```diff
@@ -8,8 +8,8 @@
 import subprocess
 import sys
 import time
-from .base_tmpl import BaseTmpl
 
+from .base_tmpl import BaseTmpl
```
Comment (gaogaotiantian) [inline tests/cmdline_tmpl.py:12]: Extra line here

File: tests/data/vdb_basic.py [hunk 8]
```diff
@@ -1,5 +1,6 @@
 # import wthell
-from viztracer import VizCounter, VizTracer, VizObject
+from viztracer import VizCounter, VizObject, VizTracer
+
```
Comment (gaogaotiantian) [inline tests/data/vdb_basic.py:3]: As we are cleaning code here, could you remove all the comments related to `wthell` or `wth`? I think those are left there when I was debugging.

Comment (gaogaotiantian) [reply tests/data/vdb_basic.py:3]: The commented exception raising as well

File: tests/test_basic.py [hunk 9]
```diff
@@ -219,14 +221,14 @@ def my_function2(n):
 
             def t1():
                 a = subprocess.run(
-                    ["ls result_my_function2*.json"], shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+                    ["ls result_my_function2*.json"], shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
```
Comment (gaogaotiantian) [inline tests/test_basic.py:224]: I don't like the comma here, let's do something like

```python
a = subprocess.run(["ls result_my_function2*.json"],
                   shell=True,
                   stdout=subprocess.PIPE,
                   stderr=subprocess.PIPE)
```

Same for the code below.

File: tests/test_cmdline.py [hunk 10]
```diff
@@ -3,16 +3,16 @@
...
 import signal
 import sys
-import re
+from contextlib import contextmanager
 from unittest.case import skipIf
+
 from .cmdline_tmpl import CmdlineTmpl
 from .package_env import package_matrix
 
-
```
Comment (gaogaotiantian) [inline tests/test_cmdline.py:15]: 2 lines here

File: tests/test_codemonkey.py [hunk 11]
```diff
@@ -22,7 +24,7 @@ def test_compile_empty_exception(self):
                 a = 3 / 0
             except Exception as e:
                 raise
-            """
+            """,
```
Comment (gaogaotiantian) [inline tests/test_codemonkey.py:27]: Just move the parenthesis up to avoid this comma.

File: tests/test_flamegraph.py [hunk 12]
```diff
@@ -20,7 +22,7 @@ def test_dump_perfetto(self):
 
         # Test if FlameGraph can handle empty tree
         sample_data["traceEvents"].append(
-            {"ph": "M", "pid": 1, "tid": 1, "name": "thread_name", "args": {"name": "MainThread"}}
+            {"ph": "M", "pid": 1, "tid": 1, "name": "thread_name", "args": {"name": "MainThread"}},
```
Comment (gaogaotiantian) [inline tests/test_flamegraph.py:25]: Move the parenthesis up to avoid this comma

File: tests/test_functree.py [hunk 13]
```diff
@@ -3,9 +3,10 @@
 
 
 import json
+
 from viztracer.functree import FuncTree
-from .base_tmpl import BaseTmpl
 
+from .base_tmpl import BaseTmpl
```
Comment (gaogaotiantian) [inline tests/test_functree.py:9]: Extra line here

File: tests/test_logsparse.py [hunk 14]
```diff
@@ -1,10 +1,10 @@
 # Licensed under the Apache License: http://www.apache.org/licenses/LICENSE-2.0
 # For details: https://github.com/gaogaotiantian/viztracer/blob/master/NOTICE.txt
 
-from .cmdline_tmpl import CmdlineTmpl
 import multiprocessing
 import os
 
+from .cmdline_tmpl import CmdlineTmpl
```
Comment (gaogaotiantian) [inline tests/test_logsparse.py:7]: Extra line here

File: tests/test_multiprocess.py [hunk 15]
```diff
@@ -1,15 +1,15 @@
...
 import multiprocessing
+import os
 import signal
+import sys
 import tempfile
 import unittest
 from unittest.case import skipIf
-from .cmdline_tmpl import CmdlineTmpl
 
+from .cmdline_tmpl import CmdlineTmpl
```
Comment (gaogaotiantian) [inline tests/test_multiprocess.py:12]: Extra line here

File: tests/test_patch.py [hunk 16]
```diff
@@ -12,6 +11,7 @@
 import unittest
 from string import Template
 
+from .cmdline_tmpl import CmdlineTmpl
```
Comment (gaogaotiantian) [inline tests/test_patch.py:14]: Extra line

File: tests/test_vizplugin.py [hunk 17]
```diff
@@ -27,12 +29,11 @@ def f(data):
...
             return {
                 "action": "handle_data",
-                "handler": f
+                "handler": f,
             }
 
-        if m_type == "command":
-            if payload["cmd_type"] == "terminate":
-                return {"success": self.terminate_well}
+        if m_type == "command" and payload["cmd_type"] == "terminate":
```
Comment (gaogaotiantian) [inline tests/test_vizplugin.py:35]: Please revert this change - this is the structure left for future development. Currently there is only one command `terminate`, but it's possible that there are more in the future.

**Review by gaogaotiantian — CHANGES_REQUESTED** [review]
I went through all the changes and left comments on the lines that I'd like some changes. Most of them were:
1. Lines between import section and the code. I think your linter likes one line between the import section and non-definition code. However, I'd like to keep the import section two lines away from any code(except for those that can be considered part of the import section).
2. Ways to avoid a trailing comma at a weird position.

Thank you for the work on this!

**Comment by gaogaotiantian** [comment]
Hi @marksmayo, are you still interested in working on this PR? If not, I can make changes based on your current code. Thanks!

**Comment by gaogaotiantian** [comment]
Hi @marksmayo , your contribution is greatly appreciated. As I'm preparing for v0.16.0 release, I want the changes in soon. So I'll merge this in a feature PR and make some changes there myself, then merge into main branch. I'll list you as the co-author. Thank you!

**Comment by marksmayo** [comment]
@gaogaotiantian apologies, I'd been locked out of my github when changing phones.  All fixed now, glad to see this merged :)

**Comment by gaogaotiantian** [comment]
@marksmayo no worries! Thanks for the contribution! I'm glad the ancient coding style issues are fixed.

**Pull request merged**